src/Raven.Server/ServerWide/Commands/ClusterTransactionCommand.cs

```diff
@@ -669,11 +672,12 @@ public static IEnumerable<SingleClusterDatabaseCommand> ReadCommandsBatch<TTrans
                         continue;
                     }
 
-                    if (take <= 0)
-                        yield break;
-
                     take--;
                     yield return result;
+
+                    readSize += reader.Size;
+                    if(readSize > maxBytesToRead)
```

Comment:
The batch size will always exceed the threshold to ensure we avoid an infinite loop when a single command exceeds the limit.